src/DataRenderer.tsx

```diff
@@ -85,33 +89,67 @@ function ExpandableObject({
   const lastIndex = data.length - 1;
 
   const onKeyDown = (e: React.KeyboardEvent<HTMLSpanElement>) => {
-    if (e.key === ' ') {
+    if (e.key === 'ArrowRight' || e.key === 'ArrowLeft') {
+      e.preventDefault();
+      setExpanded(e.key === 'ArrowRight');
+    } else if (e.key === 'ArrowUp' || e.key === 'ArrowDown') {
       e.preventDefault();
-      toggleExpanded();
+      const direction = e.key === 'ArrowUp' ? -1 : 1;
+
+      if (!outerRef.current) return;
+      const buttonElements = Array.from(
```

Comment:
also here not sure, if we really improve performance but how avoid to create array and use NodeList directly?

```js
const buttonElements = outerRef.current.querySelectorAll<HTMLElement>('[role=button]');

let currentIndex = -1;

buttonElements.forEach((el, i) => {
  if (el.tabIndex === 0) currentIndex = i;
});
if (currentIndex < 0) return;
```